packages/spacecat-shared-data-access/src/models/site/site.collection.js

```diff
@@ -71,6 +71,33 @@ class SiteCollection extends BaseCollection {
 
     return orderedSites;
   }
+
+  async addSiteCompetitorBySiteId(siteId, siteCompetitorBaseURL) {
+    const site = await this.findById(siteId);
+    if (!site) {
+      throw new DataAccessError(`Site with id ${siteId} not found`, this);
+    }
+    const currentSiteCompetitors = site.getSiteCompetitors();
+    if (!currentSiteCompetitors.includes(siteCompetitorBaseURL)) {
+      currentSiteCompetitors.push(siteCompetitorBaseURL);
```

Comment:
this will not work, you'd have to call `site.setSiteCompetitors(updatedCompetitors)` - please validate in IT test

also, you could use a set to make things easier:
```js
  const competitors = new Set(site.getSiteCompetitors());
  if (!competitors.has(siteCompetitorBaseURL)) {
    competitors.add(siteCompetitorBaseURL);
    site.setSiteCompetitors([...competitors]);
    await site.save();
  }
```